packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -71,17 +71,48 @@ export class NoSource extends BuildSource {
   }
 }
 
+/**
+ * The construction properties common to all build sources that are backed by Git.
+ */
+export interface GitBuildSourceProps extends BuildSourceProps {
+  /**
+   * The depth of history to download. Minimum value is 0.
+   * If this value is 0, greater than 25, or not provided,
+   * then the full history is downloaded with each build of the project.
+   */
+  cloneDepth?: number;
+}
+
+/**
+ * A common superclass of all build sources that are backed by Git.
+ */
+export abstract class GitBuildSource extends BuildSource {
```

Comment:
I think this is an example of over-modeling. Now what happens if we have another Git source that's not in the codebuild package, then what...

Reply:
But I was so proud of how I DRYied it up... 😭

What alternative would you propose here Elad?

Reply:
Just duplicate. Sometimes it's too DRY

Reply:
But I am okay with this too :-D

Reply:
I can't. I can't bear myself to copy & paste `cloneDepth` after I've so beautifully applied inheritance and the Decorator pattern here. My object-oriented programming education forbids me.

I'll leave this as is for now, feel free to open a separate PR changing this behind my back 😃